### PR TITLE
[doc/lc_ctrl] update wording for ext_clock_en

### PR DESCRIPTION
**Pull request by cindychip**
This PR updates lc_ctrl documentation according to issue #16529.

Signed-off-by: Cindy Chen <chencindy@opentitan.org>

### CODE REVIEW

File: hw/ip/lc_ctrl/doc/_index.md [hunk 1]
```diff
@@ -224,7 +224,7 @@ The CHECK_BYP_EN signal is only asserted when a transition command is issued.
 
 If the life cycle state is in RAW, TEST* or RMA, and if {{< regref TRANSITION_CTRL.EXT_CLOCK_EN >}} is set to one, the CLK_BYP_REQ signal is asserted in order to switch the main system clock to an external clock signal.
 This functionality is needed in certain life cycle states where the internal clock source may not be fully calibrated yet, since the OTP macro requires a stable clock frequency in order to reliably program the fuse array.
-The CLK_BYP_REQ signal is only asserted when a transition command is issued.
+The CLK_BYP_REQ signal is only asserted when a claim register ({{< regref "CLAIM_TRANSITION_IF" >}}) command is issued.
```
Comment (msfschaffner) [inline hw/ip/lc_ctrl/doc/_index.md:227]: I would reword this as follows:
```
Note that the {{< regref TRANSITION_CTRL.EXT_CLOCK_EN >}} register can only be set to one if the transition interface has been claimed via the {{< regref "CLAIM_TRANSITION_IF" >}} mutex.
```

**Review by msfschaffner — APPROVED** [review]

File: hw/ip/lc_ctrl/doc/_index.md (hunk 1 above)
Comment (cindychip) [reply hw/ip/lc_ctrl/doc/_index.md:227]: Thank you, updated!
